### PR TITLE
Upgrade to ink4

### PR DESCRIPTION
**Pull request by bobo-k2**
Contracts upgraded to Ink! 4.0
Integration tests upgraded to WeightV2 and to the latest Typechain

### CODE REVIEW

File: contracts/marketplace/Cargo.toml [hunk 1]
```diff
@@ -1,39 +1,24 @@
 [package]
 name = "marketplace"
-version = "0.2.0"
+version = "0.3.0"
```
Comment (Maar-io) [inline contracts/marketplace/Cargo.toml]: why not 0.4 :)

File: contracts/psp34/lib.rs [hunk 2]
```diff
@@ -3,56 +3,49 @@
...
-                    String::from("name").into_bytes(),
-                    String::from("TestPSP34").into_bytes(),
-                );
-                instance._set_attribute(
-                    collection_id,
-                    String::from("symbol").into_bytes(),
-                    String::from("TETS").into_bytes(),
-                );
-            })
+            let mut _instance = Self::default();
```
Comment (Maar-io) [inline contracts/psp34/lib.rs]: remove underscore since you are using this variable

File: contracts/marketplace/Cargo.toml (hunk 1 above)
Comment (bobo-k2) [reply contracts/marketplace/Cargo.toml]: Why 0.4? Because of ink! 4?

File: tests/marketplace.spec.ts [hunk 3]
```diff
@@ -334,6 +336,19 @@ describe('Marketplace tests', () => {
...
   }
+
+  function getEstimatedGas(gasRequired: Weight): WeightV2 {
+    // For some reason Typechain returns wrong type Weigh, although under the hood
+    // WeightV2 structure is stored
+    const gasRequiredV2 = gasRequired as unknown as WeightV2;
+    return api.registry.createType(
+      'WeightV2',
+      {
+        refTime: gasRequiredV2.refTime.toBn().muln(2),
```
Comment (Maar-io) [inline tests/marketplace.spec.ts:347]: try on 4.50 astar local node. It works for me without multiplying with 2

**Review by Maar-io — APPROVED** [review]
lgtm

**Pull request merged**